### DISCUSSION TITLE
OH,  ESH, and Eclipse Foundation - what are they and how do they relate

### DISCUSSION THREAD

**Discussion by bgilmer77** [openhab/openhab-docs#2050] [Team Posts]
So... newb question based on a few things I think I know, but can't pull together.  First, what I think I know is right.
- OH is an open-source project worked on by volunteers
-  Eclipse Foundation is some sort of legal entity with contributor agreements, etc. for stuff related to the world of Eclipse, whatever that is.
-  There is a software application I have on my Mac called Eclipse which appears to be an IDE.  I have messed with it a little bit, but not much.  But generally I get what an IDE is for, even though I am not a programmer.  Why this is called Eclipse, and how it relates, if at all, to the Eclipse Foundation or ESH I have no clue.
-  There is some sort of project out on GitHub called ESH.  Is this an open-source project? No clue.  Is this a paid product?  No clue.  Is OH an open-source community that has a paid component that in some way makes OH a viable "free" product, partially paid for by ESH? No clue.
-  There are some documents from the ESH repo that are carbon-copied over to the OH doc repo.  I am sure this is to avoid maintaining two copies of the same thing (avoiding versioning nightmares, etc.). But I have no idea who the ESH community is, how the doc maintainers on ESH relate to us, or whether it is fine for us to barge into their world and contribute changes in their project since our repo draws upon theirs.

Inquiring minds want to know...

**Comment by Confectrician** [comment]
Hi,

I will try to clarify it from my knowledge, so maybe i can not cover all points.

> OH is an open-source project worked on by volunteers

Correct.

> Eclipse Foundation is some sort of legal entity with contributor agreements, etc. for stuff related to the world of Eclipse, whatever that is.

Correct.

> There is a software application I have on my Mac called Eclipse which appears to be an IDE. I have messed with it a little bit, but not much. But generally I get what an IDE is for, even though I am not a programmer. Why this is called Eclipse, and how it relates, if at all, to the Eclipse Foundation or ESH I have no clue.

Eclipse IDE is -one- project (of many) which is hold/maintained by the Eclipse Foundation.
ESH is another project that is hold/maintaned by the Eclipse Foundation.

> There is some sort of project out on GitHub called ESH. Is this an open-source project? No clue. Is this a paid product? No clue. Is OH an open-source community that has a paid component that in some way makes OH a viable "free" product, partially paid for by ESH? No clue.

ESH in a Nutshell:
openHAB was a "standalone" solution in its first version.
During the development of OH2 there was the point when a part (the conceptual one) of openHAB 2 was splitted 
from the openHAB Project and got unter holding of the Eclipse Foundation as "Eclipse Smarthome".
At that Point openHAB was the first "End-User-Product" that used ESH.

So what is ESH now in short Words: ESH is a Framework that provides Concepts and an API which adress SmartHome problems.

Both of them are open source, but have different licensing einvoronments.
With the split up of ESH it was possible to make the concepts and APIs attractive for companies too.
From what i have read Paper UI was developed this way from an external company into the ESH core.

[See this post, which sums it up in one of its paragraphs.](https://community.openhab.org/t/oh2-vs-oh1-why-one-should-migrate/35968/24?u=confectrician)


> There are some documents from the ESH repo that are carbon-copied over to the OH doc repo. I am sure this is to avoid maintaining two copies of the same thing (avoiding versioning nightmares, etc.). But I have no idea who the ESH community is, how the doc maintainers on ESH relate to us, or whether it is fine for us to barge into their world and contribute changes in their project since our repo draws upon theirs.

This is the same situation as we have with binding docs.
They are copied from over there, to keep the source and docs in one repo and too reduce the needed maintenance work. (As you have mentioned already.)
If you look at the people in the openhab github organization and compare those users with the esh contributors,
you will find many known usernames.
And of course since esh is open source, contributions are welcome.
